app/ssttpcalculator/CalculatorForm.scala

```diff
@@ -42,12 +43,18 @@ object CalculatorForm {
         .verifying("ssttp.calculator.form.payment_today.amount.decimal-places", { i =>
           if (Try(BigDecimal(CurrencyUtil.cleanAmount(i))).isSuccess) BigDecimal(CurrencyUtil.cleanAmount(i)).scale <= 2 else true
         })
-        .verifying("ssttp.calculator.form.payment_today.amount.less-than-owed", i =>
-          if (i.nonEmpty && Try(BigDecimal(CurrencyUtil.cleanAmount(i))).isSuccess) BigDecimal(CurrencyUtil.cleanAmount(i)) < totalDue else true)
+        .verifying(Constraint((i: String) => if ({
```

Comment:
there are a few duplicate checks being done across the `verifying` calls - since the `verifying` calls are done in order you can perhaps remove some of the checks that have been done before, e.g. remove the `i.nonEmpty` checks since it's being checks in the first `verifying`. If you're concerned about the order not being followed then perhaps a mapping could be used instead to have greater control. We've got something like it in essttp in [this form](https://github.com/hmrc/essttp-frontend/blob/main/app/models/forms/UpfrontPaymentAmountForm.scala#L26) which refers to [this formatter](https://github.com/hmrc/essttp-frontend/blob/main/app/models/MoneyUtil.scala#L34).